### PR TITLE
Switch async void to async Task

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

**Comment by bricelam** [comment]
Missing `using System.Threading.Tasks;` in a few places...

File: test/EFCore.Tests/Storage/ExecutionStrategyTest.cs [hunk 1]
```diff
@@ -416,7 +416,7 @@ public void ExecuteAsync_Func_throws_for_an_enlisted_transaction()
             ExecuteAsync_throws_for_an_enlisted_transaction(e => e.ExecuteAsync(ct => Task.FromResult(1), CancellationToken.None));
         }
 
-        private async void ExecuteAsync_throws_for_an_enlisted_transaction(Func<ExecutionStrategy, Task> executeAsync)
+        private async Task ExecuteAsync_throws_for_an_enlisted_transaction(Func<ExecutionStrategy, Task> executeAsync)
```
Comment (Eilon) [inline test/EFCore.Tests/Storage/ExecutionStrategyTest.cs]: Nothing is awaiting this method.

File: test/EFCore.Tests/Storage/ExecutionStrategyTest.cs [hunk 2]
```diff
@@ -391,7 +391,7 @@ public void ExecuteAsync_Func_throws_for_an_ambient_transaction()
             ExecuteAsync_throws_for_an_ambient_transaction(e => e.ExecuteAsync(ct => Task.FromResult(1), CancellationToken.None));
         }
 
-        private async void ExecuteAsync_throws_for_an_ambient_transaction(Func<ExecutionStrategy, Task> executeAsync)
+        private async Task ExecuteAsync_throws_for_an_ambient_transaction(Func<ExecutionStrategy, Task> executeAsync)
```
Comment (Eilon) [inline test/EFCore.Tests/Storage/ExecutionStrategyTest.cs]: Nothing is awaiting this method.

File: samples/DbContextPooling/Program.cs [hunk 3]
```diff
@@ -101,7 +101,7 @@ private static async Task SimulateRequestsAsync(IServiceProvider serviceProvider
             }
         }
 
-        private static async void MonitorResults(TimeSpan duration, Stopwatch stopwatch)
+        private static async Task MonitorResults(TimeSpan duration, Stopwatch stopwatch)
```
Comment (Eilon) [inline samples/DbContextPooling/Program.cs]: Oh and no one is awaiting this method either.

Comment (javiercn) [reply samples/DbContextPooling/Program.cs]: I'll take a look, I just did a find & replace all over the repos

Comment (bricelam) [reply samples/DbContextPooling/Program.cs]: The EF samples may or may not be functional--most are just used to add code to the docs. 😉

**Pull request merged**